WordPress/UITestsFoundation/Screens/MySiteScreen.swift

```diff
@@ -204,9 +239,25 @@ public class MySiteScreen: ScreenObject {
 
     @discardableResult
     public func scrollToFreeToPaidPlansCard() throws -> Self {
-        let collectionView = app.collectionViews.firstMatch
-        let cardCell = collectionView.cells.containing(.other, identifier: "dashboard-free-to-paid-plans-card-contentview").firstMatch
-        cardCell.scrollIntoView(within: collectionView)
+        scrollToCard(withId: ElementStringIDs.freeToPaidPlansCardId)
+        return self
+    }
+
+    @discardableResult
+    public func scrollToPagesCard() throws -> Self {
+        scrollToCard(withId: ElementStringIDs.pagesCardId)
         return self
     }
+
+    @discardableResult
+    public func tapPagesCardHeader() throws -> PagesScreen {
+        pagesCardHeaderButton.tap()
+        return try PagesScreen()
+    }
+
+    func scrollToCard(withId id: String) {
+        let collectionView = app.collectionViews.firstMatch
+        let cardCell = collectionView.cells.containing(.any, identifier: id).firstMatch
+        app.scrollDownToElement(element: cardCell)
+    }
```

Comment:
Since there are two cards now - "Domains" and "Pages" (and more coming) - that need scrolling, I extracted the scrolling part to a separate method. The major change here is that I'm not using `scrollIntoView` extension any longer, because it appeared unstable for the case when more than one swipe is needed to scroll to the element. Instead, I'm using newly added `scrollDownToElement`.